tests/theMap.test.js

```diff
@@ -0,0 +1,60 @@
+/* eslint-disable no-undef */
+/* eslint-disable max-len */
+import React from 'react';
+import { shallow } from 'enzyme';
+import TheMap from '../components/map/index';
+
+const mockRegion = {
+  latitude: 45.492408,
+  longitude: -73.582153,
+  latitudeDelta: 0.04,
+  longitudeDelta: 0.04
+};
+
+const mockPoi = {
+  result: {
+    name: 'some name',
+    formattedAddress: 'some address',
+    geometry:
+        {
+          location: {
+            lat: 45.492409,
+            lng: -73.582153
+          },
+        }
+  }
+};
+
+beforeEach(() => {
+  getDestinationIfSet = jest.fn();
+  updateRegionCloser = jest.fn();
+});
+
+
+it('Should fit screen to updated components', () => {
+  const spy = jest.spyOn(TheMap.prototype, 'fitScreenToPath').mockImplementation(() => { return true; });
+  wrapper = shallow(<TheMap updatedRegion={mockRegion} />);
+  expect(spy).toHaveBeenCalled();
+});
+
+it('Should fetch and send the selected point of interest to the home component', async () => {
+  getDestinationIfSet = jest.fn();
+  updateRegionCloser = jest.fn();
+  component = shallow(<TheMap updatedRegion={mockRegion} getDestinationIfSet={getDestinationIfSet} updateRegionCloser={updateRegionCloser} />);
```

Comment:
Is there any benefit enzyme brings to the table that the react renderer does not for testing? (the tests tend to get the component details with the react renderer like so):
const withinBuildingComponent = renderer.create(<WithinBuilding />).getInstance();

Reply:
Yes, In this particular situation: 
I need to use shallow rendering. Howver, react-test-renderer/shallow does not call componentDidMount and it only provides top level of the component. (ie props and structure).

I specifically need shallow rendering because the instantiation of child components which comes from an API crashes the test.

It seems that the shallow renderer it the most flexible and popular technique that we can use for shallow rendering.

Thus I included Enzyme.